sdk/core/Azure.Core/src/Response{T}.cs

```diff
@@ -1,23 +1,31 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.ComponentModel;
 
 namespace Azure
 {
     public class Response<T>
     {
         private readonly Response _rawResponse;
+        private readonly Lazy<T> _value;
 
         public Response(Response response, T parsed)
         {
             _rawResponse = response;
-            Value = parsed;
+            _value = new Lazy<T>(() => parsed);
+        }
+
+        public Response(Response response, Exception exception)
+        {
+            _rawResponse = response;
+            _value = new Lazy<T>(() => throw exception);
```

Comment:
Just curious - why does this Lazy need to exist in either case? Isn't the expensive part (the resolution of T) already completed? 

Ie, why isn't .Value here just {
   if (exception != null) {
      throw exc;
  }

  return value;
}

Reply:
The idea was to postpone throwing the exception until the caller asked for _value.  We have a compiler check that _value can't be null so it has to be set to something.